ouroboros-consensus-test/src/Test/Util/ChainDB.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE FlexibleContexts #-}
+module Test.Util.ChainDB where
+
+
+import           Data.Functor.Identity (Identity)
+
+-- import           Control.Monad.Class (MonadThrow)
+import           Control.Monad.Class.MonadThrow (MonadThrow)
+import           Control.Tracer (Tracer)
+
+import           Ouroboros.Consensus.Config (TopLevelConfig,
+                     configSecurityParam)
+import           Ouroboros.Consensus.Fragment.InFuture (CheckInFuture (..))
+import qualified Ouroboros.Consensus.Fragment.Validated as VF
+import           Ouroboros.Consensus.Ledger.Extended (ExtLedgerState)
+import           Ouroboros.Consensus.Protocol.Abstract
+import           Ouroboros.Consensus.Storage.ChainDB.Impl.Args
+import           Ouroboros.Consensus.Storage.ChainDB.Impl.LgrDB (LedgerDB')
+import           Ouroboros.Consensus.Storage.ChainDB.Impl.Types
+                     (TraceEvent (..))
+import           Ouroboros.Consensus.Storage.FS.API (SomeHasFS (..))
+import qualified Ouroboros.Consensus.Storage.ImmutableDB as ImmutableDB
+import qualified Ouroboros.Consensus.Storage.LedgerDB.DiskPolicy as LedgerDB
+import qualified Ouroboros.Consensus.Storage.VolatileDB as VolatileDB
+import           Ouroboros.Consensus.Util.MonadSTM.NormalForm
+import           Ouroboros.Consensus.Util.ResourceRegistry (ResourceRegistry)
+import qualified Test.Util.FS.Sim.MockFS as Mock
+
+import           Test.Util.FS.Sim.STM (simHasFS)
+
+import           Ouroboros.Consensus.Block.Abstract
+import           Test.Util.FS.Sim.MockFS
+
+
+-- TODO: This already exists in the ThreadNet tests
+
+-- | A vector with an element for each database of a node
+--
+-- The @db@ type parameter is instantiated by this module at types for mock
+-- filesystems; either the 'MockFS' type or reference cells thereof.
+data NodeDBs db = NodeDBs
+  { nodeDBsImm :: db
+  , nodeDBsVol :: db
+  , nodeDBsLgr :: db
+  }
+
+
+emptyNodeDBs :: MonadSTM m => m (NodeDBs (StrictTVar m MockFS))
+emptyNodeDBs = NodeDBs
+  <$> uncheckedNewTVarM Mock.empty
+  <*> uncheckedNewTVarM Mock.empty
+  <*> uncheckedNewTVarM Mock.empty
+
+
+-- | 1. No integrity checking of blocks
+mkTestChainDbArgs :: (MonadThrow m, MonadSTM m, ConsensusProtocol (BlockProtocol blk))
+                  => TopLevelConfig blk
+                  -> ImmutableDB.ChunkInfo
+                  -- ^ Specifies the layout of the ImmutableDB on disk.
```

Comment:
I think this could also benefit from a good default (e.g. using `simpleChunkInfo`, and extracting `chunkSize` from the `TopLevelConfig` if `NoHardForks blk` is satisfied).